rosette/src/BaseSupp.cc

```diff
@@ -297,7 +297,9 @@ uint32_t mem_get_field(uint32_t* addr, int offset, int span, int sign) {
         /*
          * The following makes big-endian assumptions.
          */
-
+// #if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
```

Comment:
I don't know of any situation in which spans will be generated with anything other than a multiple of 8 bits, but if we're not going to have a compile error here, we should at least throw an exception.

Reply:
If there are indeed no cases where this questionable code is needed, I would agree with removing it in favor of a fatal error.  Given that there are no catch blocks in the current code, throwing an exception would just result in the program bailing out completely. We need to at least log the error before bailing out.

Reply:
This code can't possibly be running, nor doing anything. Every single processor that this code runs on is little-endian. Which, fwiw, has nothing to do with BIT order, but byte order in a multibyte thing. In general, the only time that you'd notice this would be if you are exchanging numbers over a network.

```
/tmp
leaf@lisbeth $ cat endian.cpp 
#include <stdio.h>

using namespace std;

int main(int c, char** v) {

    printf ("Endianness: ");

#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
    printf("Little endian\n");
#endif

#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
    printf("Big endian\n");
#endif

#if __BYTE_ORDER__ == __ORDER_PDP_ENDIAN__
    printf("PDP endian, wow.\n");
#endif

    return 0;
}

leaf@lisbeth $ ./endian 
Endianness: Little endian
```

Reply:
However, don't comment dead code, just delete it.

Reply:
**Done**